packages/profile-sync-controller/src/shared/encryption/utils.ts

```diff
@@ -14,3 +16,35 @@ export const bytesToUtf8 = (byteArray: Uint8Array) => {
 export const stringToByteArray = (str: string) => {
   return new TextEncoder().encode(str);
 };
+
+export const areAllUInt8ArraysUnique = (arrays: Uint8Array[]) => {
+  const seen = new Set<string>();
+  for (const arr of arrays) {
+    const str = arr.toString();
+    if (seen.has(str)) {
+      return false;
+    }
+    seen.add(str);
+  }
+  return true;
```

Comment:
Could this be simplified?
```ts
const strSet = new Set(arrays.map(arr => arr.toString()))
return strSet.size === arrays.length
```

But yeah your func if more performant (since it bails earlier)